pkgs/servers/monitoring/mtail/default.nix

```
@@ -1,22 +1,24 @@
-{ lib, fetchFromGitHub, gotools, buildGoPackage }:
+{ lib, fetchFromGitHub, buildGoModule }:
 
-buildGoPackage rec {
+buildGoModule rec {
   pname = "mtail";
-  version = "3.0.0-rc4";
-  goPackagePath = "github.com/google/mtail";
+  version = "3.0.0-rc34";
 
   src = fetchFromGitHub {
     owner = "google";
     repo = "mtail";
-    rev = "v${version}";
-    sha256 = "1n7pqvid48ayn15qfpgpbsx0iqg24x08wphzpc08mlfw47gq7jg3";
+    rev = "7b4270cb6e8b6ed1f2febe48e370cb68d38681f9";
+    sha256 = "0hrbv6dhiqwl1kyrjmwwplq5wig1yak0gql6w4cnac0060ba7r92";
   };
 
-  buildInputs = [ gotools ];
-  goDeps = ./deps.nix;
-  patches = [ ./fix-gopath.patch ];
-  preBuild = "go generate -x ./go/src/github.com/google/mtail/vm/";
+  modSha256 = "1iipmja3zcmpc5mg9q41p26wrccl9z937y40aqi0cyld7y94lafc";
+  subPackages = [ "cmd/mtail" ];
 
+  preBuild = "go generate -x ./internal/vm/";
```

I'd use a multiline nix string, for the sake of consistency across Nixpkgs:

```suggestion
  preBuild = ''
    go generate -x ./internal/vm/
  '';
```